pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -107,8 +170,8 @@ func TestReconcile(t *testing.T) {
 						t.Errorf("Expected actions to be logged in the buildclient, got none")
 					}
 					build := client.Actions()[0].(ktesting.CreateAction).GetObject().(*buildv1alpha1.Build)
-					if d := cmp.Diff(build.Spec, buildSpec); d != "" {
-						t.Errorf("Expected created resource to be %v, but was %v", build.Spec, buildSpec)
+					if d := cmp.Diff(build.Spec, tc.wantedBuildSpec); d != "" {
```

Comment:
im curious on your thoughts here @dlorenc - i feel like the set of situations this one particular test is supposed to handle are growing over time, what about creating a separate function specifically for testing the case where we have replacements in a task?

Reply:
OK, cleaned up the test a bit. I actually ended up leaving this one in the same test, but reorganized the tests around it.

This test actually only had one other tc before this, but it was complicated enough to make that hard to see.

Now there's one test for times a build should have been created (with and without templating), and one for error cases.